pkg/provider/azure_controller_common.go

```diff
@@ -529,71 +545,64 @@ func (c *controllerCommon) GetDiskLun(diskName, diskURI string, nodeName types.N
 	return -1, provisioningState, fmt.Errorf("%s for disk %s", consts.CannotFindDiskLUN, diskName)
 }
 
-// SetDiskLun find unused luns and allocate lun for every disk in disksPendingAttach map.
-// Return err if not enough luns are found.
-func (c *controllerCommon) SetDiskLun(nodeName types.NodeName, disksPendingAttach map[string]*AttachDiskOptions) error {
+// SetDiskLun finds an unused lun and allocates lun for disk pending attach.
+// Return err if not available lun cannot be found.
+func (c *controllerCommon) SetDiskLun(nodeName types.NodeName, diskPendingAttach *AttachDiskParams) error {
+	uri := diskPendingAttach.diskURI
+	opt := diskPendingAttach.options
+	if opt == nil {
+		return fmt.Errorf("unexpected nil pointer in diskPendingAttach(%v)", diskPendingAttach)
+	}
+
+	entry, _ := c.tempLunMap.LoadOrStore(string(nodeName), new(TempLunMapping))
```

Comment:
In order to prevent allocating a new `TempLunMapping` structure on every `SetDiskLun` call, it's best to `Load` first and then `LoadOrStore` if a previous entry does not exist, e.g.

```go
entry, ok := c.tempLunMap.Load(string(nodeName))
if !ok {
    entry, _ := c.tempLunMap.LoadOrStore(string(nodeName), new(TempLunMapping))
}
```

Reply:
I think what Ed is referring to here is that we always allocate new `TempLunMapping` structure to pass to `LoadOrStore` call even when allocating one is not necessary. By checking `Load` first, we can avoid in most cases unnecessarily allocating `TempLunMapping` struct.